### PR TITLE
Cif2mtz column labels

### PR DESCRIPTION
**Pull request by Oeffner**
This branch unifies a newer version of parsing cif reflection data in iotbx.cif.miller_array_builder with the existing version implemented by @rjgildea . It uses regular expressions for parsing cif reflection column labels to allow more flexibility when interpreting columns as specific types data types.
It also fixes a bug where some phase and amplitudes columns like _refln.pdbx_FWT and _refln.pdbx_PHWT were not always grouped together after parsing. Together with minor changes in mmtbx\command_line\cif_as_mtz.py this allows converting a cif file with the columns:

_refln.status 
_refln.pdbx_r_free_flag 
_refln.F_meas_au 
_refln.F_meas_sigma_au 
_refln.F_calc_au 
_refln.phase_calc 
_refln.pdbx_HL_A_iso 
_refln.pdbx_HL_B_iso 
_refln.pdbx_HL_C_iso 
_refln.pdbx_HL_D_iso 
_refln.pdbx_FWT 
_refln.pdbx_PHWT 
_refln.pdbx_DELFWT 
_refln.pdbx_DELPHWT 
_refln.fom 

into an mtz file with column groups labbelled as:
'FOBS,SIGFOBS', 'FOM', 'HLA,HLB,HLC,HLD', 'R-free-flags', 'R-free-flags-1', FC,PHIFC', 'FWT,PHWT', 'DELFWT,PHDELWT'

The current version on master branch fails to include the last three column groups'FC,PHIFC', 'FWT,PHWT', 'DELFWT,PHDELWT'. This is currently the case for the cif structure factor file with PDB ID: 6CXO.
Cif column labels, scale_group_code=#, crystal_id=# and wavelength_id=#, will only be included in miller_array labels if their absence would otherwise lead to ambiguities of data columns, i.e. if there are more than one wavelength recorded, say. These labels are now appended to the miller_array labels rather than prepended. 

Some regression tests have been updated to reflect the additional columns produced by cif_as_mtz.py as well as what is incorrect parsing of some columns (e.g. exercise_11() of phenix_regression/cif_as_mtz/tst_cif_as_mtz.py). 
The test iotbx/regression/tst_export_scalepack_unmerged.py has been updated to reflect that wavelength_id is not inlcuded in file names when there is no ambiguity.
The branch cif2mtz_column_labels in the phenix_regression module therefore must also be merged into master in concert with merging this branch here into master.

### CODE REVIEW

**Comment by bkpoon** [comment]
We're sorting out the current failures. Let's rebase your branch once they're fixed.

**Comment by bkpoon** [comment]
@Oeffner Can you pull in the latest changes?

File: iotbx/cif/builders.py [hunk 1]
```diff
@@ -333,263 +337,28 @@ class miller_array_builder(crystal_symmetry_builder):
...
-            labels=list(
-              OrderedSet(plus_array.info().labels+minus_array.info().labels))))
-          array.set_observation_type(plus_array.observation_type())
-          self._arrays.setdefault(key, array)
-
-    if len(self._arrays) == 0:
-      raise CifBuilderError("No reflection data present in cif block")
-
+  def guess_observationtype(self, labl):
+    for okey in list(self.observation_types.keys()):
```
Comment (bkpoon) [inline iotbx/cif/builders.py:346]: These keys probably do not need to be in a list since the dictionary is not modified in the loop.

**Review by bkpoon — APPROVED** [review]
I'll run some additional tests. This can be in the April 2021 release (#610) if this is merged by Friday.

**Review by olegsobolev — APPROVED** [review]
Assuming merging with fixes for cif_as_mtz in phenix_regression (there seems to be a branch there).

**Comment by bkpoon** [comment]
The tests for `dials`, `dxtbx`, and `xia2` behave the same with these changes. The tests were done with Python 3.7 on CentOS 7 with these commands,
```
libtbx.pytest -n 72 --regression  # dxtbx
```
[dxtbx_before.log](https://github.com/cctbx/cctbx_project/files/6444507/dxtbx_before.log)
[dxtbx_after.log](https://github.com/cctbx/cctbx_project/files/6444508/dxtbx_after.log)

```
libtbx.pytest -n 72 --regression --runslow  # dials
```
[dials_before.log](https://github.com/cctbx/cctbx_project/files/6444509/dials_before.log)
[dials_after.log](https://github.com/cctbx/cctbx_project/files/6444510/dials_after.log)

```
libtbx.pytest -n 72 --regression --regression-full # xia2
```
[xia2_before.log](https://github.com/cctbx/cctbx_project/files/6444505/xia2_before.log)
[xia2_after.log](https://github.com/cctbx/cctbx_project/files/6444506/xia2_after.log)

and the `dials_regression` and `xfel_regression` modules configured.

**Pull request merged**